src/tactic/core.lean

```diff
@@ -1156,28 +1157,57 @@ meta def dependent_pose_core (l : list (expr × expr)) : tactic unit := do
 /-- Like `mk_local_pis` but translating into weak head normal form before checking if it is a `Π`.
 -/
 meta def mk_local_pis_whnf (e : expr) (md := semireducible) : tactic (list expr × expr) := do
-(expr.pi n bi d b) ← whnf e md | return ([], e),
-p ← mk_local' n bi d,
-(ps, r) ← mk_local_pis (expr.instantiate_var b p),
-return ((p :: ps), r)
-
-/-- Changes `(h : ∀xs, ∃a:α, p a) ⊢ g` to `(d : ∀xs, a) (s : ∀xs, p (d xs) ⊢ g` and
- `(h : ∀xs, p xs ∧ q xs) ⊢ g` to `(d : ∀xs, p xs) (s : ∀xs, q xs) ⊢ g` 
- `choose1` returns the second local constant it introduces. -/
-meta def choose1 (h : expr) (data : name) (spec : name) : tactic expr := do
+  expr.pi n bi d b ← whnf e md | return ([], e),
+  p ← mk_local' n bi d,
+  (ps, r) ← mk_local_pis (expr.instantiate_var b p),
+  return ((p :: ps), r)
+
+/-- Given `α : Sort u`, `nonemp : nonempty α`, `p : α → Prop`, a context of local variables
+`ctxt`, and a pair of an element `val : α` and `spec : p val`, produce another pair `val', spec'`
```

Comment:
```suggestion
`ctxt`, and a pair of an element `val : α` and `spec : p val`, 
`mk_sometimes u α nonemp p ctx (val, spec)` produces another pair `val', spec'`
```

Reply:
And similar below. Without dependencies, variable names don't show up in the docs, and writing like this is clearer.